pkg/operator/csidriveroperator/util.go

```diff
@@ -63,3 +68,42 @@ func reportUpdateEvent(recorder events.Recorder, obj runtime.Object, originalErr
 		recorder.Eventf(fmt.Sprintf("%sUpdated", gvk.Kind), "Updated %s:\n%s", resourcehelper.FormatResourceForCLIWithNamespace(obj), strings.Join(details, "\n"))
 	}
 }
+
+func checkDeploymentHealth(ctx context.Context, c appsclientv1.DeploymentsGetter, d *appsv1.Deployment) error {
+	d, err := c.Deployments(d.Namespace).Get(ctx, d.Name, metav1.GetOptions{})
```

Comment:
Something very similar is done in CVO and OLM to make sure the Degraded condition is set when something is wrong with the operator Deployment.